src/ConnectionManager.ts

```diff
@@ -41,38 +41,43 @@ export class ConnectionManager {
       }
     }
 
-    const connector = this.buildConnector(providerType, chainIdToConnect)
-    connector.on(ConnectorEvent.Deactivate, this.handleWeb3ReactDeactivate)
-    const { provider, account }: ConnectorUpdate = await connector.activate()
+    try {
+      const connector = this.buildConnector(providerType, chainIdToConnect)
+      connector.on(ConnectorEvent.Deactivate, this.handleWeb3ReactDeactivate)
 
-    if (providerType === ProviderType.MAGIC) {
-      connector.on(ConnectorEvent.Update, ({ chainId }) => {
-        if (chainId) {
-          this.setConnectionData(providerType, chainId)
-        }
-      })
-    }
+      const { provider, account } = await connector.activate()
 
-    let chainId = chainIdToConnect
+      if (providerType === ProviderType.MAGIC) {
+        connector.on(ConnectorEvent.Update, ({ chainId }) => {
+          if (chainId) {
+            this.setConnectionData(providerType, chainId)
+          }
+        })
+      }
 
-    // We need to return the correct current chain id for the injected providers
-    if (providerType === ProviderType.INJECTED) {
-      const currentChainIdHex = (await provider.request({
-        method: 'eth_chainId'
-      })) as string
-      chainId = currentChainIdHex
-        ? (parseInt(currentChainIdHex, 16) as ChainId)
-        : chainId
-    }
+      let chainId = chainIdToConnect
 
-    this.connector = connector
-    this.setConnectionData(providerType, chainId)
+      // We need to return the correct current chain id for the injected providers
+      if (providerType === ProviderType.INJECTED) {
+        const currentChainIdHex = (await provider.request({
+          method: 'eth_chainId'
+        })) as string
+        chainId = currentChainIdHex
+          ? (parseInt(currentChainIdHex, 16) as ChainId)
+          : chainId
+      }
 
-    return {
-      provider: ProviderAdapter.adapt(provider),
-      providerType,
-      account: account || '',
-      chainId
+      this.connector = connector
+      this.setConnectionData(providerType, chainId)
+
+      return {
+        provider: ProviderAdapter.adapt(provider),
+        providerType,
+        account: account || '',
+        chainId
+      }
+    } catch (error) {
+      throw error
```

Comment:
Does this try catch make ay sense? It is just throwing the error without catching anything